### PR TITLE
Fixed #8 exception when Response is empty

### LINKED ISSUE #8: Exception when Response is Empty
See https://github.com/google/agera/issues/144 for ref. as reported by @mahmoud-elnaggar

### PR DESCRIPTION
**Pull request by drakeet**
In addition:
- Updated agera to v1.3.0, retrofit to v2.2.0, support libs to v25

### CODE REVIEW

**Comment by drakeet** [comment]
Related issue: https://github.com/google/agera/issues/144

**Comment by drakeet** [comment]
@ernstsson According to Agera's design, `Result.absent().succeeded()` return `false`, but for example, the DELETE request is actually a success. 

So, I am thinking of putting the body into the `Result` is appropriate?
And how about `Result.success(value)` accept null value?

**Comment by drakeet** [comment]
@ernstsson There is also a workaround solution:
If `response.body()` is `null`, put an empty String into Result: `Result.success ("")`.

What about it?

**Comment by drakeet** [comment]
@mahmoud-elnaggar

**Comment by ghost** [comment]
Yeah, using an empty string in this case would be preferable.

Agera itself is def. encouraging the use of the null object pattern over passing null

**Comment by drakeet** [comment]
But a empty string cannot be cast to the `T` 🤔

**Comment by ghost** [comment]
Oh, sorry about that, didn't read the code very closely, I admit.

I see two different solutions;
1. return an absent, and not only check succeeded() but also isAbsent, on the client side.
or;
2. provide a parameter as a null object, an instance of T.

Since I haven't used this extension myself, I'm not entirely sure what would be preferable here.

**Comment by drakeet** [comment]
Maybe we could create a class called `NullBody`, then we could use it to wrap the null body.

**Comment by drakeet** [comment]
It is late at night in China, so the problem will continue to be resolved tomorrow.

**Comment by drakeet** [comment]
@ernstsson I thought for a while, these two solutions are good, I am currently favored in the solution one...

**Comment by mahmoud-elnaggar** [comment]
I vote for first solution , it is more reasonable

File: agera-call-adapter/src/test/java/me/drakeet/retrofit2/adapter/agera/SupplierTest.java [hunk 1]
```diff
@@ -147,12 +151,27 @@
...
+
+
+    @Test public void shouldReturnSuccessWithNullBodyResponse() throws Exception {
+        MockResponse mockResponse = new MockResponse().setStatus(STATUS_NO_CONTENT);
+        assertNull(mockResponse.getBody());
+        server.enqueue(mockResponse);
+
+        Supplier<Result<Response<Void>>> supplier = service.responseOfDeleteXXX();
+        Result<Response<Void>> responseResult = supplier.get();
+        assertTrue(responseResult.succeeded());
```
Comment (drakeet) [inline agera-call-adapter/src/test/java/me/drakeet/retrofit2/adapter/agera/SupplierTest.java:171]: OK, if we expect succeeded result, we can use `Result<Response<Void>>`.

**Comment by drakeet** [comment]
Thank you very much for the help! We now accept the `absentIfNull`.

**Comment by drakeet** [comment]
I have uploaded archives, waiting for Maven Central to take effect, it may take several hours.

**Comment by mahmoud-elnaggar** [comment]
Thanks very much for fixing this issue

**Pull request merged**